apm-agent-plugins/apm-jms-plugin/apm-jms-plugin-base/src/main/java/co/elastic/apm/agent/jms/JmsMessagePropertyAccessor.java

```diff
@@ -78,6 +79,8 @@ public void setHeader(String headerName, String headerValue, Message message) {
         headerName = jmsifyHeaderName(headerName);
```

Comment:
IIRC, @eyalkoren mentioned that in this specific issue, the header might have already been set.
By checking for the presence of the header, we may prevent the exception.

```suggestion
        headerName = jmsifyHeaderName(headerName);
        if (getFirstHeader(headerName, message) != null) {
            return;
        }
```

Reply:
good suggestion, I've added a few test for that also.